cirq-core/cirq/ops/linear_combinations.py

```diff
@@ -839,10 +839,14 @@ def __pow__(self, exponent: int):
         if exponent == 0:
             return PauliSum(value.LinearDict({frozenset(): 1 + 0j}))
         if exponent > 0:
+            remainder = PauliSum.from_pauli_strings([PauliString(coefficient=1)])
             base = self.copy()
-            for _ in range(exponent - 1):
-                base *= base
-            return base
+            while exponent > 0:
+                if exponent & 1:
+                    remainder = remainder * base
+                base = base * base
```

Comment:
the last `base` value in the loop would be unused.
can you please move the `exponent` check before the base squaring?

Reply:
Thanks for the observations @pavoljuhas, worked on it in my latest submission.